src/Features/LanguageServer/Protocol/Handler/InlayHint/InlayHintHandler.cs

```diff
@@ -0,0 +1,134 @@
+        {
+            var document = context.GetRequiredDocument();
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            var textSpan = ProtocolConversions.RangeToTextSpan(request.Range, text);
+
+            var inlineHintService = document.GetRequiredLanguageService<IInlineHintsService>();
+            var options = _optionsService.GetInlineHintsOptions(document.Project.Language);
+
+            var hints = await inlineHintService.GetInlineHintsAsync(document, textSpan, options, displayAllOverride: false, cancellationToken).ConfigureAwait(false);
+
+            using var _ = ArrayBuilder<LSP.InlayHint>.GetInstance(hints.Length, out var inlayHints);
+            var syntaxVersion = await document.GetSyntaxVersionAsync(cancellationToken).ConfigureAwait(false);
+            var inlayHintCache = context.GetRequiredLspService<InlayHintCache>();
+
+            // Store the members in the resolve cache so that when we get a resolve request for a particular
+            // member we can re-use the inline hint.
+            var resultId = inlayHintCache.UpdateCache(new InlayHintCache.InlayHintCacheEntry(hints, request.TextDocument, syntaxVersion));
+
+            for (var i = 0; i < hints.Length; i++)
+            {
+                var hint = hints[i];
+                var (label, leftPadding, rightPadding) = Trim(hint.DisplayParts);
+                var linePosition = text.Lines.GetLinePosition(hint.Span.Start);
+                var kind = hint.Ranking == InlineHintsConstants.ParameterRanking
+                    ? InlayHintKind.Parameter
+                    : InlayHintKind.Type;
+
+                // TextChange is calculated at the same time as the InlineHint,
+                // so it should not need to be resolved.
+                /*TextEdit[]? textEdits = null;
+                if (hint.ReplacementTextChange.HasValue)
+                {
+                    var textEdit = ProtocolConversions.TextChangeToTextEdit(hint.ReplacementTextChange.Value, text);
+                    textEdits = new TextEdit[] { textEdit };
+                }*/
+
+                var inlayHint = new LSP.InlayHint
+                {
+                    Position = ProtocolConversions.LinePositionToPosition(linePosition),
+                    Label = label,
+                    Kind = kind,
+                    TextEdits = null,
+                    ToolTip = null,
+                    PaddingLeft = leftPadding,
+                    PaddingRight = rightPadding,
+                    Data = new InlayHintResolveData(resultId, i)
+                };
+
+                inlayHints.Add(inlayHint);
+            }
+
+            return inlayHints.ToArray();
+        }
+
+        private static (string label, bool leftPadding, bool rightPadding) Trim(ImmutableArray<TaggedText> taggedTexts)
```

Comment:
can you doc a quick explanation of the goal here?   E.g. what needs to be trimmed, maybe with an example